[improve][doc] improve Pulsar function document (#18601)
apache/pulsar: site2/docs/functions-package-python.md

1. Write a Python function.

   ```python
   from pulsar import Function #  import the Function module from Pulsar

>    

Reviewer: extra spaces ?

Author: Same reason as below.

Reviewer: See my comment below

>    

Reviewer: ```suggestion

```

   # The classic ExclamationFunction that appends an exclamation at the end
   # of the input
   class ExclamationFunction(Function):
      def __init__(self):
         pass

>       

Reviewer: extra spaces ?

Author: Yes, python default indent is 4 spaces, so I add extra spaces. I've tested it, users can copy & paste this code snippet to a file directly.

Reviewer: No, I mean there are spaces in L26 that I believe should not be there since the line is empty

Author: Understand. I have removed them.

>       

Reviewer: ```suggestion

```

2. Install a Python client. The implementation of a Python function depends on the Python client. 

   ```bash
   pip install pulsar-client==2.10.0
   ```
   

>    And install protobuf tools to generate proto file:

Reviewer: ```suggestion
   And install protobuf tools to generate the proto files:
```
